src/vulkan-renderer/wrapper/window.cpp

```diff
@@ -17,14 +17,26 @@ Window::Window(const std::string &title, const std::uint32_t width, const std::u
 
     spdlog::debug("Creating window.");
 
-    m_window = glfwCreateWindow(width, height, title.c_str(), nullptr, nullptr);
+    GLFWmonitor *monitor = nullptr;
+    if (m_mode != Mode::WINDOWED) {
+        monitor = glfwGetPrimaryMonitor();
+        if (m_mode == Mode::WINDOWED_FULLSCREEN) {
+            auto mode = glfwGetVideoMode(monitor);
+            m_width = mode->width;
+            m_height = mode->height;
+        }
+    }
+
+    m_window = glfwCreateWindow(m_width, m_height, title.c_str(), monitor, nullptr);
 
     if (!m_window) {
         throw std::runtime_error("Error: glfwCreateWindow failed for window " + title + " !");
     }
 }
 
-Window::Window(Window &&other) noexcept : m_window(std::exchange(other.m_window, nullptr)) {}
+Window::Window(Window &&other) noexcept
+    : m_window(std::exchange(other.m_window, nullptr)), m_width(std::exchange(other.m_width, 0)),
+      m_height(std::exchange(other.m_height, 0)), m_mode(other.m_mode) {}
```

Comment:
@yeetari is exchange usefull for the simple width, height values?

Reply:
I think we should just copy by value for simple stuff, right?

Reply:
Yeah the width and height don't actually need to be exchanged here, only `m_window` matters.